### PR TITLE
DE-971 - Compatibility with the new jenkins

### PR DESCRIPTION
**Pull request by andresmoschini**
Related to https://github.com/FromDoppler/doppler-jenkins-swarm/pull/5

- Adding a suffix (_empty string_ in the old Jenkins and `-new` in the new Jenkins) to the generated Docker images. It allows pushing the images to Docker Hub without collisions with the official ones.

- Using builtin gitlint or dockerized version depending on availability

-  Adding a new symlink `doppler-jenkins-ci.groovy` to identify that the new Jenkins should listen to this repository.

-  ~~Adding a new symlink `doppler-jenkins-testi.groovy` to identify that the test instance of the new Jenkins should listen to this repository.~~

- ~Using `TESTING-JENKINS` branch in the same way as `INT`. It will allow to run builds with the [testing Jenkins server](https://jenkins.fromdoppler.net/test/) in place with the [real one](https://jenkins.fromdoppler.net/ci/).~

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -58,7 +58,7 @@ pipeline {
                     }
                     steps {
                         withDockerRegistry(credentialsId: "${DOCKER_CREDENTIALS_ID}", url: "") {
-                            sh 'sh build-n-publish.sh --image=${DOCKER_IMAGE_NAME} --commit=${GIT_COMMIT} --name=pr-${CHANGE_ID}'
+                            sh 'sh build-n-publish.sh --image=${DOCKER_IMAGE_NAME}${PACKAGE_SUFFIX} --commit=${GIT_COMMIT} --name=pr-${CHANGE_ID}'
```
Comment (RodrigoPereyraDiaz) [inline Jenkinsfile:61]: Where is defined the value of `PACKAGE_SUFFIX` ?

What do you think about append the suffix when set the value of `DOCKER_IMAGE_NAME` ? That will allow change only one line

[the remark this answers is not in the document]
Comment (andresmoschini) [reply Jenkinsfile:61]: It is [an environment variable](https://github.com/FromDoppler/doppler-jenkins-swarm/pull/5/files#diff-9ca5f821aefab32ac273eddc5c117aa07024d3f26a6b6b0c4a5e7e2c105e880fR28), you can see a description in https://github.com/FromDoppler/doppler-jenkins-swarm/pull/5

TL;DR: I am using it to avoid collisions with the artifacts generated in the new Jenkins. By the moment, it adds the suffix `-new`, but in the future it will be an empty string.

Comment (andresmoschini) [reply Jenkinsfile:61]: In my mind, it is explicit and clear, and I already did it in this way in a lot of repositories, so, I prefer to keep it as it is here. But you can do it in that way in your repositories.

Take into account that it is a temporal patch.

File: Dockerfile [hunk 2]
```diff
@@ -14,7 +14,9 @@ RUN shellcheck -e SC1091,SC1090 ./*.sh
 FROM mcr.microsoft.com/dotnet/sdk:7.0.202-bullseye-slim AS restore
 WORKDIR /src
 COPY ./*.sln ./
-COPY */*.csproj ./
+# Using `Doppler.` prefix to avoid docker confuse symlink with directories
+# "ERROR: error from sender: readdir: readdirent Jenkinsfile: not a directory"
+COPY Doppler.*/*.csproj ./
```
Comment (andresmoschini) [inline Dockerfile:19]: Hi @RodrigoPereyraDiaz!

After applying [this change](https://github.com/FromDoppler/doppler-jenkins/pull/30/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R36), this build started to fail.

It seems that current docker version is confusing the symlinks files `.doppler-ci` and `doppler-jenkins-ci.groovy` as directories and the line `COPY */*.csproj ./` was failing  with this error:

image

This change fixes the issue, but we will need to apply the change in all repositories. Another option could be to copy explicitly each .csproj file to his folder.

I do not want to downgrade docker version, and I do not want to remove build neither because before that change I was getting this warning:

```
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
Install the buildx component to build images with BuildKit:
https://docs.docker.com/go/buildx/
```

What do you think?

Comment (andresmoschini) [reply Dockerfile:19]: Finally, I decided to remove buildx because it is breaking many builds.

https://github.com/FromDoppler/doppler-jenkins/pull/33

File: Jenkinsfile (hunk 1 above)
Comment (andresmoschini) [reply Jenkinsfile:61]: I applied your suggestion in [MSEditor](https://github.com/MakingSense/MSEditor/pull/1489/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8R25) and [Forms](https://github.com/MakingSense/doppler-forms/pull/424/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8R51).

You are right, it looks better.
